manual/manual/cmds/comp.etex

```diff
@@ -188,6 +188,28 @@ to find by itself the ".cmo" file that implements a module with a
 given name: it relies instead on the user providing the list of ".cmo"
 files by hand.
 
+The {\em load path} is a list of directories and single files maintained by the
+compiler to look up, in order, for files needed to compile and/or link programs.
+
+By default the load path consists of the current directory and the standard
+library directory, but new entries can be added by using the "-I" and "-inc"
+flags.
+
+A directory can be added to the load path using "-I <dir>". Whenever the
+compiler needs to look up a file given by a relative path it will look up inside
+"<dir>". This includes the case of looking up top-level modules mentioned in the
+program text, but also relative paths passed on the command-line.
+
+A single file can also be added to the load path using "-inc <file>". This
+option is equivalent to "-I d" where "d" is a fresh directory containing only a
+single file with the same basename as "<file>". This option is meant mostly for
```

Comment:
> This option is equivalent to "-I d" where "d" is a fresh directory containing only a single file with the same basename as "<file>".

This doesn't actually make sense to me. The sentence suggests that `-inc foo.cmi` is equivalent to `-I /tmp/<random>/foo.cmi`, which it surely isn't (`tmp/<random>/foo.cmi` does not exist!). Can we explain what `-inc` does without trying to piggy-back on `-I`? (Shouldn't it go in the `-inc` documentation rather than here?)

Reply:
> > This option is equivalent to "-I d" where "d" is a fresh directory containing only a single file with the same basename as "".
> 
> This doesn't actually make sense to me. The sentence suggests that `-inc foo.cmi` is equivalent to `-I /tmp/<random>`, which it surely isn't (what is the content of `tmp/<random>/foo.cmi`?).

Indeed, the statement is only true if `foo.cmi` exists.

> Can we explain what `-inc` does without trying to piggy-back on `-I`? (Shouldn't it go in the `-inc` documentation rather than here?)

I will give it a try. The slightly tedious thing to explain is the fact that `-inc` is only used to resolve "basename" files, which sort of follows logically from the interpretation in terms of `-I`.